src/main/java/com/linecorp/armeria/common/metrics/DropwizardMetricConsumer.java

```diff
@@ -33,8 +33,16 @@ public DropwizardMetricConsumer(MetricRegistry metricRegistry, String metricName
     }
 
     @Override
+    public void invocationStart(Scheme scheme, String hostname, String path, Optional<String> method) {
+        final String metricName = MetricRegistry.name(metricNamePrefix, method.orElse("__unknown__"));
+        final DropwizardRequestMetrics metrics = getRequestMetrics(metricName);
+        metrics.markStart();
+    }
+
+    @Override
     public void invocationComplete(Scheme scheme, int code, long processTimeNanos, int requestSize,
```

Comment:
More things to consider:

- `MetricConsumer.invocationComplete()` is not invoked if `ClientCodec.decodeResponse()` is not called.
  - This can happen when a decorator like a circuit breaker decides not to make an actual invocation.
- As @taicki mentioned, the current metrics does not tell us if the failure was caused by a client (e.g. circuit broken) or by a server (e.g. server responded an error response.)
  - How do we represent client-induced failures? Special error codes?
- Armeria currently do not provide a way for a user to define what success/failure is. It's hard-coded - `>= 400`
  - Should Armeria determine if an invocation was successful or not? Is it the responsibility of the monitoring system?
    - If Armeria should, should it be customizable? (e.g. if we add HBase/Redis client/server in the future, they will use different code values.)